src/content/developers/docs/smart-contracts/testing/index.md

```diff
@@ -1,37 +1,217 @@
+
+Functional testing covers three methods: **unit testing**, **integration testing**, and **system testing**.
+
+- **Unit testing**
+
+Unit testing involves testing individual components in a smart contract for correctness. A unit test is simple, quick to run, and provides a clear idea of what went wrong if the test fails. 
+
+Unit tests are crucial for smart contract development, especially if you need to add new logic to the code. You can verify the behavior of each feature and confirm that it executes as intended.
+
+Running a unit test often requires creating *assertions*—simple, informal statements specifying requirements for a smart contract. Unit testing can then be used to test each assertion and see if it holds true under execution. 
+
+Examples of contract-related assertions include:
+
+i. "Only the admin can pause the contract" 
+
+ii. "Non-admins cannot mint new tokens" 
+
+iii. "The contract reverts on errors" 
+
+- **Integration testing**
+
+Integration testing is a level higher than unit testing on the testing hierarchy. In integration testing, individual components of the smart contract are tested together. 
+
+This approach detects errors arising from interactions between different components of a contract or across multiple contracts. You should use this method if you have a complex contract with multiple functions or one that interfaces with other contracts. 
+
+Integration testing can be useful for ensuring that things like [inheritance](https://docs.soliditylang.org/en/v0.8.12/contracts.html#inheritance) and dependency injection work properly. 
+
+- **System testing**
+
+System testing is the final phase of functional testing for smart contracts. A system evaluates the smart contract as one fully integrated product to see if it performs as specified in the technical requirements.
+
+You can think of this stage as checking the end-to-end flow of your smart contract from a user’s point of view. A good way to perform system testing on a smart contract is to deploy it on a production environment, such as a [testnet](/developers/docs/networks/#ethereum-testnets) or [development network](/developers/docs/development-networks/).  
+
+Here, end-users can perform trial runs and report any issues with the contract’s business logic and overall functionality. System testing is important because you cannot change code once the contract is deployed in the main EVM environment. 
 
-**Waffle -** **_A framework for advanced smart contract development and testing (based on ethers.js)._**
+System testing can also cover separate testing measures, such as load testing and regression testing. A load test (also called a stress test) assesses the program's ability to perform optimally under real-time loads, while regression tests ensure changes made during development (after previous tests) didn't introduce new defects in the final contract. 
```

Comment:
```suggestion
```
I suggest removing this paragraph. Load testing isn't really relevant to smart contracts. All tests also handle some amount of regression testing.